sdk/core/src/http.rs

```diff
@@ -0,0 +1,44 @@
+pub struct Request {
+    inner: http::Request<bytes::Bytes>,
+}
+
+impl Request {
+    /// Get the inner http::Request object, replacing it
+    /// with an empty one.
+    /// Note: this method will soon be replaced
+    pub fn take_inner(&mut self) -> http::Request<bytes::Bytes> {
+        std::mem::replace(&mut self.inner, http::Request::new(bytes::Bytes::new()))
+    }
+
+    pub fn body<T: serde::Serialize>(
+        &mut self,
+        body: T,
+    ) -> Result<(), Box<dyn std::error::Error + Sync + Send>> {
+        let b = self.inner.body_mut();
+        *b = crate::to_json(&body)?;
+        Ok(())
+    }
+}
+
+impl From<http::Request<bytes::Bytes>> for Request {
+    fn from(inner: http::Request<bytes::Bytes>) -> Self {
+        Self { inner }
+    }
+}
+
+pub struct Response {
```

Comment:
In our various SDKs we also have `Response<T>` or something like that, which clients would return to wrap the actual response model, but with response information, e.g.:

```csharp
Response<KeyVaultKey> response = client.GetKey("name");
KeyVaultKey key = response.Value;
Response rawResponse = response.GetRawResponse();
```

Is this something we should do now, or can we later?

Also, is `inner` just the body, or the entire response? Might be good to call it `body` if the former.

Reply:
How other SDKs handle returning the stream? Without reading the stream first it is impossible to convert it into a struct (`KeyVaultKey` in your example).  Is there a specific exception for blob storage?